### DISCUSSION TITLE
How to add SAP BTP icons & shapes as default libraries

### DISCUSSION THREAD

**Discussion by fabianleh** [jgraph/drawio#4186] [General]
* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.
Hi all,
can you explain how and what is needed to have a "deep" integration (via More Shapes) into draw.io for SAP Business Technology Platform icons and shapes. Example: 
<img width="508" alt="image" src="https://github.com/jgraph/drawio/assets/91141653/a50fe11f-3de5-4a84-8095-2a98a7757b7b">

It would be great to have these icons & shapes as default, currently we have 12 libraries in total which we want to integrate.
Draw.io is already used internally at SAP and externally like in the [SAP community](https://community.sap.com/t5/sap-teched-blog-posts/business-technology-platform-btp-solution-diagrams-icons-as-seen-at-teched/ba-p/13577704).

thx and best,
Fabian

**Comment by fdicarlo** [comment]
Hi,

Maybe this article will help: https://drawio-app.com/blog/draw-io-training-exercise-9-create-your-own-custom-library/

I just found as I would like to add my updated Azure stencils :)

**Reply by fabianleh** [reply]
Hi @fdicarlo , how to create a custom library is already known we're using draw.io some years already at SAP.
<img width="411" alt="image" src="https://github.com/jgraph/drawio/assets/91141653/6764e9f5-b56a-4ce4-9950-cada98084c9f">


The main question is for us what is needed/ required to get a "deep" integration into draw.io:
<img width="407" alt="image" src="https://github.com/jgraph/drawio/assets/91141653/100a35a0-7372-42c9-ab52-0f30b3e6088a">

Are you aware of the process here?

thx and best,
Fabian

**Comment by davidjgraph** [comment]
For a set of shapes to be embedded within the app itself, there needs to be strong, general, public demand for the set.

The shape set in the app is a substantial proportion of the total size when distributed, there needs to be interest across a range of groups, not just from the one company that uses the set. Some sets are in now because we didn't have size considerations when they were introduced and I wouldn't accept them today. Some current sets are candidates for removal, but this is very difficult to do if they are used in diagrams.

We had a user from largeCorp recently complaining (somewhat unprofessionally) that he couldn't get his set in. The issue was simply that largeCorp were interested in the set and literally nobody else was. He complained that AWS seemed to be supported. The thing with AWS is it's one of the most used domain-specific icons sets (possibly the most used), it's impossible to have a diagramming tool without it.

The initial suggestion would be to create a feature request with a link to the set and the license for that set. We then watch votes (thumbs up) to see if there is interest (and we know whether it's just SAP employees voting :) ).

Shape libraries that don't get into the core can be linked into the app quite simply. Look at the README for https://github.com/jgraph/drawio-libs. You can create links that include the library on loading. The JSON configuration can be changed in both desktop and confluence to auto-load the libraries.

Just note that if we start to get agreements sent to us from commercial in a company, or custom licenses, we run away. Microsoft did this with the azure set, so we said no until they changed it to something simple (which they did).

**Reply by fabianleh** [reply]
Hi @davidjgraph ,

1st of all thx for the explanation and totally agree the demand should not driven internally ;)
The Integration has definitely a public demand especially as the SAP Community/SAP Ecosystem is using draw.io often, see also this [blog](https://community.sap.com/t5/technology-blogs-by-members/create-sap-btp-solution-diagrams-with-draw-io-diagrams-net/ba-p/13561722) to create such BTP Solution Diagrams.
We also delivered a [session](https://github.com/SAP-samples/teched2023-XP286v/tree/main) (recording [here](https://www.youtube.com/watch?v=FCBvnc5does)) at SAP TechEd this year how to make use of SAP/BTP shapes by using draw.io.

But sure we will also provide the shapes for draw.io soon at GitHub, as suggested we will create a feature request and in parallel we will ask the SAP Community to vote on this to get this embedded in draw.io soon.

**Comment by nagesh-caparthy** [comment]
Hi @fabianleh,

Where can I find an icon for Joule service?

Regards,
Nagesh

**Reply by fabianleh** [reply]
Hi @nagesh-caparthy , not sure if this is the right discussion for that!, but we plan to release the BTP service icons (public) for draw.io later this month. best,
Fabian

**Comment by fabianleh** [comment]
@davidjgraph as we plan to have the the release from the SAP BTP Icons/Shapes etc., I just saw also this discussion which is also SAP related https://github.com/jgraph/drawio/issues/3067 
For the rollout we will also ask for vote of the SAP stencils (SAP BTP Icons&Shapes) than via a new feature request?
